bitnami/minio/templates/deployment-standalone.yaml

```diff
@@ -0,0 +1,131 @@
+{{- if eq .Values.mode "standalone" }}
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ include "minio.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "minio.name" . }}
+    helm.sh/chart: {{ include "minio.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "minio.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "minio.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- if or .Values.podAnnotations }}
+      annotations: {{ toYaml .Values.podAnnotations | nindent 8 }}
+      {{- end }}
+    spec:
+      {{- include "minio.imagePullSecrets" . | nindent 6 }}
+      affinity:
+      {{- if eq .Values.antiAffinity "hard" }}
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: {{ include "minio.name" . }}
+                  app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- else if eq .Values.antiAffinity "soft" }}
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            podAffinityTerm:
+              topologyKey: kubernetes.io/hostname
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: {{ include "minio.name" . }}
+                  app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- end }}
+      {{- if .Values.nodeAffinity }}
+        nodeAffinity: {{ toYaml .Values.nodeAffinity | nindent 10 }}
+      {{- end }}
+      {{- if .Values.securityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.securityContext.fsGroup }}
+      {{- end }}
+      containers:
+      - name: "minio"
+        image: {{ include "minio.image" . }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.securityContext.enabled }}
+        securityContext:
+          runAsUser: {{ .Values.securityContext.runAsUser }}
+        {{- end }}
+        env:
+        {{- if .Values.image.debug }}
+        - name: BITNAMI_DEBUG
+          value: "true"
+        {{- end }}
+        - name: MINIO_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              {{- if .Values.existingSecret }}
```

Comment:
Move this logic to a helper function. This is useful when an umbrella chart uses this one as a subchart. Example, postgresql